File: ruby/lib/cekat_event_sdk/acknowledgement.rb
# frozen_string_literal: true

module CekatEventSdk
  # Cekat accepted the event for asynchronous processing. This does not confirm durable
  # storage, identity resolution, delivery completion, or analytics availability.
  Acknowledgement = Data.define(:success, :message, :event_key, :validated_properties, :raw_body)
end

File: dotnet/src/Cekat.EventSdk/Acknowledgement.cs
namespace Cekat.EventSdk;

/// <summary>
/// Cekat accepted the event for asynchronous processing. This does not confirm durable storage,
/// identity resolution, or analytics availability.
/// </summary>
/// <param name="Success">Always <see langword="true"/>.</param>
/// <param name="Message">The server message.</param>
/// <param name="EventKey">The accepted event key.</param>
/// <param name="ValidatedProperties">The property names the server validated.</param>
/// <param name="RawBody">The response body.</param>
public sealed record Acknowledgement(
    bool Success,
    string Message,
    string EventKey,
    IReadOnlyList<string> ValidatedProperties,
    string RawBody);
